lib/wibox/hierarchy.lua

```diff
@@ -51,6 +51,15 @@ local function hierarchy_new(redraw_callback, layout_callback, callback_arg)
         end
         layout_callback(result, callback_arg)
     end
+    function result._emit_recursive(_, name, ...)
+        local cur = result
```

Comment:
Nope, wrong self. `self` is the widget, we need the hierarchy. I added an assert instead